File: go/lang/panic/panic.go
package panic

import (
    "github.com/otterop/otterop/go/lang/string"
)


func NewPanic() *Panic {
    this := new(Panic)
    return this
}

type Panic struct {
}




func IndexOutOfBounds(message *string.String)  {
    panic(message.Unwrap())
}

File: c/otterop/lang/panic.c
#include <otterop/lang/panic.h>
#include <gc.h>
#include <assert.h>

void otterop_lang_Panic_index_out_of_bounds(otterop_lang_String_t *message) {
    assert(!*otterop_lang_String_unwrap(message));
}

void otterop_lang_Panic_invalid_operation(otterop_lang_String_t *message) {
    assert(!*otterop_lang_String_unwrap(message));
}
